recipes/nmos-cpp/all/conanfile.py

```diff
@@ -0,0 +1,222 @@
+import json
+import os
+import re
+from conans import ConanFile, CMake, tools
+
+required_conan_version = ">=1.33.0"
+
+class NmosCppConan(ConanFile):
+    name = "nmos-cpp"
+    description = "An NMOS C++ Implementation"
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/sony/nmos-cpp"
+    topics = ("NMOS")
+
+    settings = "os", "compiler", "build_type", "arch"
+    # for now, no "shared" option support
+    options = {
+        "fPIC": [True, False],
+        "with_dnssd": ["mdnsresponder", "avahi"],
+    }
+    # "fPIC" is handled automatically by Conan, injecting CMAKE_POSITION_INDEPENDENT_CODE
+    default_options = {
+        "fPIC": True,
+        "with_dnssd": "mdnsresponder",
+    }
+
+    # wrapper CMakeLists.txt to call conan_basic_setup()
+    exports_sources = ["CMakeLists.txt"]
+    # use cmake_find_package_multi and prefer config-file packages
+    generators = "cmake", "cmake_find_package_multi"
+
+    _cmake = None
+
+    # for out-of-source build, cf. wrapper CMakeLists.txt
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+        if self.settings.os == "Macos":
+            del self.options.with_dnssd
+        elif self.settings.os == "Linux":
+            self.options.with_dnssd = "avahi"
+        elif self.settings.os == "Windows":
+            self.options.with_dnssd = "mdnsresponder"
+
+    def requirements(self):
+        # for now, consistent with project's conanfile.txt
+        self.requires("boost/1.76.0")
+        self.requires("cpprestsdk/2.10.18")
+        self.requires("websocketpp/0.8.2")
+        self.requires("openssl/1.1.1k")
+        self.requires("json-schema-validator/2.1.0")
+        if self.options.get_safe("with_dnssd") == "mdnsresponder":
+            self.requires("mdnsresponder/878.200.35")
+            #self.options["mdnsresponder"].with_opt_patches = True
+        elif self.options.get_safe("with_dnssd") == "avahi":
+            self.requires("avahi/0.8")
+
+    def build_requirements(self):
+        self.build_requires("cmake/[>3.17]")
```

Comment:
Use the latest please :)

Reply:
Most CMake-based recipes don't even have a `build_requires` so I figured it was best to base it on the actual requirement, especially as the patch version available from the cmake recipe seems to change, old ones are commented out?

Reply:
This (needing a different cmake then the one installed) has come up in the past and we usually pin down the latest (or older) version required.

They are removed from CCI to help with build times... CMake is long to compile and has lots of versions

Reply:
Build times is what I assumed. But why set it up for a future build break, and mean that end users think they need CMake 3.21.x when they don't? I understand defaulting to the latest available version for `requirements` but not for this `build_requirements`.

Reply:
What is the minimum required CMake version? Unless it is very, very recent, CMake should not be added as a build_requirement at all

Reply:
CMake 3.17. The previous build failed due to CMake 3.16 on some of the runners.